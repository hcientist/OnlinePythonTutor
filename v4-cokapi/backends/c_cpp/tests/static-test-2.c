static int x = 123;
void foo() {
  static int x=0;
  x+=10;
}

int main() {
  static int x = 456;
  int y = 789;
  foo();
  foo();
  foo();
}
