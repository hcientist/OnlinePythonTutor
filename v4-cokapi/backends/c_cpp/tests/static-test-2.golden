{
  "code": "static int x = 123;\nvoid foo() {\n  static int x=0;\n  x+=10;\n}\n\nint main() {\n  static int x = 456;\n  int y = 789;\n  foo();\n  foo();\n  foo();\n}", 
  "trace": [
    {
      "event": "step_line", 
      "func_name": "main", 
      "globals": {
        "x": [
          "C_DATA", 
          "0x601038", 
          "int", 
          123
        ]
      }, 
      "heap": {}, 
      "line": 7, 
      "ordered_globals": [
        "x"
      ], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x (static 0x60103C)": [
              "C_DATA", 
              "0x60103C", 
              "int", 
              456
            ], 
            "y": [
              "C_DATA", 
              "0xFFF000104", 
              "int", 
              "<UNINITIALIZED>"
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 7, 
          "ordered_varnames": [
            "y", 
            "x (static 0x60103C)"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "step_line", 
      "func_name": "main", 
      "globals": {
        "x": [
          "C_DATA", 
          "0x601038", 
          "int", 
          123
        ]
      }, 
      "heap": {}, 
      "line": 9, 
      "ordered_globals": [
        "x"
      ], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x (static 0x60103C)": [
              "C_DATA", 
              "0x60103C", 
              "int", 
              456
            ], 
            "y": [
              "C_DATA", 
              "0xFFF00010C", 
              "int", 
              "<UNINITIALIZED>"
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 9, 
          "ordered_varnames": [
            "y", 
            "x (static 0x60103C)"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "step_line", 
      "func_name": "main", 
      "globals": {
        "x": [
          "C_DATA", 
          "0x601038", 
          "int", 
          123
        ]
      }, 
      "heap": {}, 
      "line": 10, 
      "ordered_globals": [
        "x"
      ], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x (static 0x60103C)": [
              "C_DATA", 
              "0x60103C", 
              "int", 
              456
            ], 
            "y": [
              "C_DATA", 
              "0xFFF00010C", 
              "int", 
              789
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 9, 
          "ordered_varnames": [
            "y", 
            "x (static 0x60103C)"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "call", 
      "func_name": "foo", 
      "globals": {
        "x": [
          "C_DATA", 
          "0x601038", 
          "int", 
          123
        ]
      }, 
      "heap": {}, 
      "line": 4, 
      "ordered_globals": [
        "x"
      ], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x (static 0x60103C)": [
              "C_DATA", 
              "0x60103C", 
              "int", 
              456
            ], 
            "y": [
              "C_DATA", 
              "0xFFF00010C", 
              "int", 
              789
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": false, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 10, 
          "ordered_varnames": [
            "y", 
            "x (static 0x60103C)"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }, 
        {
          "encoded_locals": {
            "x (static 0x601044)": [
              "C_DATA", 
              "0x601044", 
              "int", 
              0
            ]
          }, 
          "frame_id": "0xFFF0000F0", 
          "func_name": "foo", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 4, 
          "ordered_varnames": [
            "x (static 0x601044)"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_0xFFF0000F0"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "return", 
      "func_name": "foo", 
      "globals": {
        "x": [
          "C_DATA", 
          "0x601038", 
          "int", 
          123
        ]
      }, 
      "heap": {}, 
      "line": 5, 
      "ordered_globals": [
        "x"
      ], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x (static 0x60103C)": [
              "C_DATA", 
              "0x60103C", 
              "int", 
              456
            ], 
            "y": [
              "C_DATA", 
              "0xFFF00010C", 
              "int", 
              789
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": false, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 10, 
          "ordered_varnames": [
            "y", 
            "x (static 0x60103C)"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }, 
        {
          "encoded_locals": {
            "x (static 0x601044)": [
              "C_DATA", 
              "0x601044", 
              "int", 
              10
            ]
          }, 
          "frame_id": "0xFFF0000F0", 
          "func_name": "foo", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 5, 
          "ordered_varnames": [
            "x (static 0x601044)"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_0xFFF0000F0"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "step_line", 
      "func_name": "main", 
      "globals": {
        "x": [
          "C_DATA", 
          "0x601038", 
          "int", 
          123
        ]
      }, 
      "heap": {}, 
      "line": 11, 
      "ordered_globals": [
        "x"
      ], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x (static 0x60103C)": [
              "C_DATA", 
              "0x60103C", 
              "int", 
              456
            ], 
            "y": [
              "C_DATA", 
              "0xFFF00010C", 
              "int", 
              789
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 11, 
          "ordered_varnames": [
            "y", 
            "x (static 0x60103C)"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "call", 
      "func_name": "foo", 
      "globals": {
        "x": [
          "C_DATA", 
          "0x601038", 
          "int", 
          123
        ]
      }, 
      "heap": {}, 
      "line": 4, 
      "ordered_globals": [
        "x"
      ], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x (static 0x60103C)": [
              "C_DATA", 
              "0x60103C", 
              "int", 
              456
            ], 
            "y": [
              "C_DATA", 
              "0xFFF00010C", 
              "int", 
              789
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": false, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 11, 
          "ordered_varnames": [
            "y", 
            "x (static 0x60103C)"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }, 
        {
          "encoded_locals": {
            "x (static 0x601044)": [
              "C_DATA", 
              "0x601044", 
              "int", 
              10
            ]
          }, 
          "frame_id": "0xFFF0000F0", 
          "func_name": "foo", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 4, 
          "ordered_varnames": [
            "x (static 0x601044)"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_0xFFF0000F0"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "return", 
      "func_name": "foo", 
      "globals": {
        "x": [
          "C_DATA", 
          "0x601038", 
          "int", 
          123
        ]
      }, 
      "heap": {}, 
      "line": 5, 
      "ordered_globals": [
        "x"
      ], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x (static 0x60103C)": [
              "C_DATA", 
              "0x60103C", 
              "int", 
              456
            ], 
            "y": [
              "C_DATA", 
              "0xFFF00010C", 
              "int", 
              789
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": false, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 11, 
          "ordered_varnames": [
            "y", 
            "x (static 0x60103C)"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }, 
        {
          "encoded_locals": {
            "x (static 0x601044)": [
              "C_DATA", 
              "0x601044", 
              "int", 
              20
            ]
          }, 
          "frame_id": "0xFFF0000F0", 
          "func_name": "foo", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 5, 
          "ordered_varnames": [
            "x (static 0x601044)"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_0xFFF0000F0"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "step_line", 
      "func_name": "main", 
      "globals": {
        "x": [
          "C_DATA", 
          "0x601038", 
          "int", 
          123
        ]
      }, 
      "heap": {}, 
      "line": 12, 
      "ordered_globals": [
        "x"
      ], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x (static 0x60103C)": [
              "C_DATA", 
              "0x60103C", 
              "int", 
              456
            ], 
            "y": [
              "C_DATA", 
              "0xFFF00010C", 
              "int", 
              789
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 12, 
          "ordered_varnames": [
            "y", 
            "x (static 0x60103C)"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "call", 
      "func_name": "foo", 
      "globals": {
        "x": [
          "C_DATA", 
          "0x601038", 
          "int", 
          123
        ]
      }, 
      "heap": {}, 
      "line": 4, 
      "ordered_globals": [
        "x"
      ], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x (static 0x60103C)": [
              "C_DATA", 
              "0x60103C", 
              "int", 
              456
            ], 
            "y": [
              "C_DATA", 
              "0xFFF00010C", 
              "int", 
              789
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": false, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 12, 
          "ordered_varnames": [
            "y", 
            "x (static 0x60103C)"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }, 
        {
          "encoded_locals": {
            "x (static 0x601044)": [
              "C_DATA", 
              "0x601044", 
              "int", 
              20
            ]
          }, 
          "frame_id": "0xFFF0000F0", 
          "func_name": "foo", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 4, 
          "ordered_varnames": [
            "x (static 0x601044)"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_0xFFF0000F0"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "return", 
      "func_name": "foo", 
      "globals": {
        "x": [
          "C_DATA", 
          "0x601038", 
          "int", 
          123
        ]
      }, 
      "heap": {}, 
      "line": 5, 
      "ordered_globals": [
        "x"
      ], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x (static 0x60103C)": [
              "C_DATA", 
              "0x60103C", 
              "int", 
              456
            ], 
            "y": [
              "C_DATA", 
              "0xFFF00010C", 
              "int", 
              789
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": false, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 12, 
          "ordered_varnames": [
            "y", 
            "x (static 0x60103C)"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }, 
        {
          "encoded_locals": {
            "x (static 0x601044)": [
              "C_DATA", 
              "0x601044", 
              "int", 
              30
            ]
          }, 
          "frame_id": "0xFFF0000F0", 
          "func_name": "foo", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 5, 
          "ordered_varnames": [
            "x (static 0x601044)"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "foo_0xFFF0000F0"
        }
      ], 
      "stdout": ""
    }, 
    {
      "event": "return", 
      "func_name": "main", 
      "globals": {
        "x": [
          "C_DATA", 
          "0x601038", 
          "int", 
          123
        ]
      }, 
      "heap": {}, 
      "line": 13, 
      "ordered_globals": [
        "x"
      ], 
      "stack_to_render": [
        {
          "encoded_locals": {
            "x (static 0x60103C)": [
              "C_DATA", 
              "0x60103C", 
              "int", 
              456
            ], 
            "y": [
              "C_DATA", 
              "0xFFF00010C", 
              "int", 
              789
            ]
          }, 
          "frame_id": "0xFFF000110", 
          "func_name": "main", 
          "is_highlighted": true, 
          "is_parent": false, 
          "is_zombie": false, 
          "line": 13, 
          "ordered_varnames": [
            "y", 
            "x (static 0x60103C)"
          ], 
          "parent_frame_id_list": [], 
          "unique_hash": "main_0xFFF000110"
        }
      ], 
      "stdout": ""
    }
  ]
}

